src/hooks/EtherPrice.ts

```diff
@@ -1,22 +1,27 @@
-import { BigNumber } from '@ethersproject/bignumber'
+import { V1UserContext } from 'contexts/v1/userContext'
 import { V1ContractName } from 'models/v1/contracts'
-import { useCallback, useState } from 'react'
+import { useContext, useEffect } from 'react'
 import { fromWad } from 'utils/formatNumber'
-
-import useContractReader from './v1/contractReader/ContractReader'
+import { useContractReadValue } from './ContractReader'
 
 export function useEtherPrice() {
-  const [price, setPrice] = useState<number>(0)
+  const { contracts } = useContext(V1UserContext)
 
-  useContractReader({
+  const { refetchValue, value: price } = useContractReadValue({
+    contracts,
     contract: V1ContractName.Prices,
     functionName: 'getETHPriceFor',
     args: ['1'], // $1 USD
-    callback: useCallback(
-      (val?: BigNumber) => setPrice(parseFloat(fromWad(val))),
-      [setPrice],
-    ),
+    formatter: val => {
+      if (!val) return
+      return parseFloat(fromWad(val))
+    },
+  })
+
+  useEffect(() => {
+    const timer = setInterval(refetchValue, 5000)
+    return () => clearInterval(timer)
   })
 
-  return price
+  return price ?? 0
```

Comment:
any reason for defaulting to 0? undefined might be nicer to be more clear, or avoid any accidental divide-by-0 funny business

Reply:
This is what is currently set as the default, but agreed it could cause issues. Might be worth doing a quick audit of its usage to see if being undefined might cause any unexpected results

Reply:
ohh yep, I see now. probably better to return undefined, but that can be a separate PR (also pretty sure i wrote the original 0 value being returned)